src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -10,12 +10,66 @@
 
 #define THIS ((EnIk*)thisx)
 
+typedef enum {
+    /* 0 */ IK_TYPE_SILVER,
+    /* 1 */ IK_TYPE_BLACK,
+    /* 2 */ IK_TYPE_WHITE
+} EnIkType;
+
 void EnIk_Init(Actor* thisx, GlobalContext* globalCtx);
 void EnIk_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void EnIk_Update(Actor* thisx, GlobalContext* globalCtx);
 void EnIk_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-#if 0
+void EnIk_Thaw(EnIk* this, GlobalContext* globalCtx);
+s32 EnIk_IsChangingAction(EnIk* this, GlobalContext* globalCtx);
+s32 EnIk_ChooseAttack(EnIk* this);
+void EnIk_CheckActions(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupIdle(EnIk* this);
+void EnIk_Idle(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupWalk(EnIk* this);
+void EnIk_WalkTowardsPlayer(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupRun(EnIk* this);
+void EnIk_RunTowardsPlayer(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupVerticalAttack(EnIk* this);
+void EnIk_VerticalAttack(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupTakeOutAxe(EnIk* this);
+void EnIk_TakeOutAxe(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupHorizontalDoubleAttack(EnIk* this);
+void EnIk_HorizontalDoubleAttack(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupSingleHorizontalAttack(EnIk* this);
+void EnIk_SingleHorizontalAttack(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupEndHorizontalAttack(EnIk* this);
+void EnIk_EndHorizontalAttack(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupBlock(EnIk* this);
+void EnIk_Block(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupReactToAttack(EnIk* this, s32 arg1);
+void EnIk_ReactToAttack(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupDeath(EnIk* this);
+void EnIk_Death(EnIk* this, GlobalContext* globalCtx);
+void EnIk_Frozen(EnIk* this, GlobalContext* globalCtx);
+void EnIk_SetupCutscene(EnIk* this);
+void EnIk_PlayCutscene(EnIk* this, GlobalContext* globalCtx);
+void EnIk_UpdateDamage(EnIk* this, GlobalContext* globalCtx);
+void EnIk_ArmorDestroyed(EnIk* this, GlobalContext* globalCtx);
+
+typedef struct {
+    Gfx* unk00;
+    s16 unk04;
+} EnIkUnkStruct;
+
+EnIkUnkStruct D_8092BFA0[] = {
+    { object_ik_DL_00CF08, 0x0000 }, { object_ik_DL_00A5D8, 0x0000 }, { object_ik_DL_00A6D0, 0x7FFF },
+    { object_ik_DL_00A820, 0x4000 }, { object_ik_DL_00A780, 0xC000 }, { ((void*)0), 0x4000 },
+    { ((void*)0), 0xC000 },
```

Comment:
NULL. These also look like rotations at first glance, you can probably work out what they are from looking at the displaylists.